upstairs/src/lib.rs

```diff
@@ -553,6 +553,100 @@ where
                 })
                 .await?
             }
+            IOop::ExtentClose {
+                dependencies: _,
+                extent: _,
+            } => {
+                // This command should never exist on the upstairs side.
+                // We only construct it for downstairs IO, after
+                // receiving ExtentLiveClose from the upstairs.
+                panic!("[{}] Received illegal IOop::ExtentClose", client_id);
+            }
+            IOop::ExtentFlushClose {
+                dependencies,
+                extent,
+                flush_number,
+                gen_number,
+                source_downstairs,
+                repair_downstairs,
+            } => {
+                // TODO: Add dtrace probe point
+                if source_downstairs == client_id {
+                    fw.send(Message::ExtentLiveFlushClose {
+                        upstairs_id: u.uuid,
+                        session_id: u.session_id,
+                        job_id: *new_id,
+                        dependencies: dependencies.clone(),
+                        extent_id: extent,
+                        flush_number,
+                        gen_number,
+                    })
+                    .await?
+                } else if repair_downstairs.contains(&client_id) {
+                    // We are the downstairs being repaired, so just close.
+                    fw.send(Message::ExtentLiveClose {
+                        upstairs_id: u.uuid,
+                        session_id: u.session_id,
+                        job_id: *new_id,
+                        dependencies: dependencies.clone(),
+                        extent_id: extent,
+                    })
+                    .await?
+                } else {
+                    // We need to maybe send this job to the downstairs as
```

Comment:
Does this need an XXX or TODO?

Reply:
fixed in 02d694383914e8a615080888c1b1bcf9de1be8a3